interface/src/components/swap/from-section.tsx

```diff
@@ -49,7 +49,7 @@ export const FromSection = (props: Props) => {
     if (!token) {
       return
     }
-    onInputChange(tokenBalance.divideByDecimals(token.decimals).div(2).format())
+    onInputChange(tokenBalance.divideByDecimals(token.decimals).div(2).format(6))
```

Comment:
It somewhat solves the problem but isn't bulletproof. For example, what if a token has `4` decimals only? `.format(6)` may still result in the decimals error.


What if we did something like this?

```
tokenBalance.div(2).abs().divideByDecimals(token.decimals).format()
```

Note: we'll need to implement `.abs()`.

Reply:
Updated!